airbyte-integrations/connectors/destination-mssql-v2/src/main/kotlin/io/airbyte/integrations/destination/mssql/v2/convert/AirbyteValueToStatement.kt

```diff
@@ -0,0 +1,223 @@
+                if (
+                    sqlType in
+                        setOf(
+                            Types.DATE,
+                            Types.TIME,
+                            Types.TIME_WITH_TIMEZONE,
+                            Types.TIMESTAMP,
+                            Types.TIMESTAMP_WITH_TIMEZONE
+                        )
+                ) {
+                    val coercedValue = valueCoercingMapper.map(value, type)
+                    if (coercedValue.second.isEmpty()) {
+                        when (coercedValue.first) {
+                            is DateValue -> setAsDateValue(idx, coercedValue.first as DateValue)
+                            is TimeWithTimezoneValue ->
+                                setAsTime(idx, coercedValue.first as TimeWithTimezoneValue)
+                            is TimeWithoutTimezoneValue ->
+                                setAsTime(idx, coercedValue.first as TimeWithoutTimezoneValue)
+                            is TimestampWithTimezoneValue ->
+                                setAsTimestamp(
+                                    idx,
+                                    coercedValue.first as TimestampWithTimezoneValue
+                                )
+                            is TimestampWithoutTimezoneValue ->
+                                setAsTimestamp(
+                                    idx,
+                                    coercedValue.first as TimestampWithoutTimezoneValue
+                                )
+                            else -> throw IllegalArgumentException("$value isn't a $type")
+                        }
+                    } else {
+                        throw IllegalArgumentException("$value isn't a $type")
+                    }
+                } else {
+                    throw IllegalArgumentException("$value isn't a $type")
+                }
+            }
+        }
+
+        private fun PreparedStatement.setAsTime(idx: Int, value: TimeWithTimezoneValue) {
+            setTime(idx, Time.valueOf(value.value.toLocalTime()))
+        }
+
+        private fun PreparedStatement.setAsTime(idx: Int, value: TimeWithoutTimezoneValue) {
+            setTime(idx, Time.valueOf(value.value))
+        }
+
+        private fun PreparedStatement.setAsTimestamp(idx: Int, value: TimestampWithTimezoneValue) {
+            setTimestamp(idx, Timestamp.valueOf(value.value.toLocalDateTime()))
+        }
+
+        private fun PreparedStatement.setAsTimestamp(
+            idx: Int,
+            value: TimestampWithoutTimezoneValue
+        ) {
+            setTimestamp(idx, Timestamp.valueOf(value.value))
+        }
+
+        private fun createSimpleUnionObject(value: AirbyteValue): ObjectValue {
+            val unionType = value.toTypeName()
+            return ObjectValue.from(mapOf("type" to StringValue(unionType), unionType to value))
+        }
+
+        private fun createUnionObject(type: UnionType, value: AirbyteValue): AirbyteValue =
```

Comment:
why do we need to do this? I think it would be easier to just do a blind serializeToJson for any field in a Union-typed column (which is what we do for most existing destinations, and also for the upcoming s3-data-lake destination)

Reply:
I wasn't aware this object behavior was optional at the time of the implementation, so MSSQL has the full on complex object/union support.

Reply:
welp. and I guess we want to wait for johnny's types refactor before making more changes here?

(we should _definitely_ call this out in our beta comms, i.e. that union types are expected to look different in the near future)